client/my-sites/site-settings/form-general.jsx

```diff
@@ -286,71 +357,73 @@ const FormGeneral = React.createClass( {
 						<FormRadio
 							name="blog_public"
 							value="-1"
-							checked={ -1 === parseInt( this.state.blog_public, 10 ) }
+							checked={ -1 === parseInt( fields.blog_public, 10 ) }
 							onChange={ this.handleRadio }
-							disabled={ this.state.fetchingSettings }
+							disabled={ isRequestingSettings }
 							onClick={ this.onRecordEvent( 'Clicked Site Visibility Radio Button' ) } />
-						<span>{ this.translate( 'Private' ) }</span>
+						<span>{ translate( 'Private' ) }</span>
 						<FormSettingExplanation isIndented>
-							{ this.translate( 'Your site is only visible to you and users you approve.' ) }
+							{ translate( 'Your site is only visible to you and users you approve.' ) }
 						</FormSettingExplanation>
 					</FormLabel>
 				}
 
 			</FormFieldset>
 		);
-	},
+	}
 
-	handleAmpToggle() {
-		this.setState( { amp_is_enabled: ! this.state.amp_is_enabled }, () => {
+	handleAmpToggle = () => {
+		const { fields, updateFields } = this.props;
+		updateFields( { amp_is_enabled: ! fields.amp_is_enabled }, () => {
 			this.submitForm();
-			this.onRecordEvent( 'Clicked AMP Toggle' );
+			this.onRecordEvent( 'Clicked AMP Toggle' )();
```

Comment:
This syntax is a bit awkward to me (the immediate calling of the function we've created in `onRecordEvent`). Instead, should we calling directly to `this.props.recordGoogleEvent` ? We could create a `partial` or arrow function in `mapDispatchToProps` to apply the first argument consistently, or an instance function `recordEvent` (changing `onRecordEvent` to `return () => this.recordEvent( eventAction );`).